### PR TITLE
[IE][VPU] Move Myriad QueryNetwork to a separate file

### PR DESCRIPTION
**Pull request by adudchen**
-29576

### CODE REVIEW

**Comment by Maxim-Doronin** [comment]
Why didn't you place this code into 'vpu/common'? We've discussed it offline

**Comment by adudchen** [comment]
> Why didn't you place this code into 'vpu/common'? We've discussed it offline

**Comment by adudchen** [comment]
> Why didn't you place this code into 'vpu/common'? We've discussed it offline

This code uses functions from FrontEnd, so I thought it would be better to place it here

**Comment by Maxim-Doronin** [comment]
> > Why didn't you place this code into 'vpu/common'? We've discussed it offline
> 
> This code uses functions from FrontEnd, so I thought it would be better to place it here

I'll try to clarify how we can avoid using functions from GT and move `getQueryNetwork` to the common part of the VPU code base
First, you can get converted CNNNetwork on the plugin side and then pass it in `getQueryNetwork`
Secondly, you can implement `getSupportedLayers` method in the Frontend and pass the result container with supported layers to `getQueryNetwork`

File: inference-engine/src/vpu/myriad_plugin/myriad_plugin.cpp [hunk 1]
```diff
@@ -83,153 +81,14 @@ QueryNetworkResult Engine::QueryNetwork(
...
-        for (auto& node : function->get_ops()) {
-            originalOps.emplace(node->get_friendly_name());
-        }
-
         auto clonedNetwork = cloneNetwork(network);
         auto convertedNetwork = vpu::FrontEnd::convertNetwork(*clonedNetwork);
+        
+        auto stageBuilder = std::make_shared<vpu::StageBuilder>();
+        auto frontEnd = std::make_shared<vpu::FrontEnd>(stageBuilder, GetCore());
+        auto supportedLayers = frontEnd->getSupportedLayers();
```
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/myriad_plugin/myriad_plugin.cpp:89]: Can we re-use this method?
```
std::set<std::string> getSupportedLayers(
        const ie::ICNNNetwork& network,
        Platform platform,
        const CompilationConfig& config,
        const Logger::Ptr& log,
        const ie::ICore* core) 
```

File: inference-engine/src/vpu/myriad_plugin/myriad_plugin.cpp [hunk 2]
```diff
@@ -82,160 +80,23 @@ QueryNetworkResult Engine::QueryNetwork(
...
-                if (!InferenceEngine::details::contains(supported, node->output(0).get_target_inputs().begin()->get_node()->get_friendly_name())) {
-                    supported.erase(node->get_friendly_name());
-                }
-            } else if (ngraph::op::is_output(node)) {
-                if (!InferenceEngine::details::contains(supported, node->input_values().begin()->get_node()->get_friendly_name())) {
-                    supported.erase(node->get_friendly_name());
-                }
-            }
-        }
+        auto supportedLayers = getSupportedLayers(network,
```
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/myriad_plugin/myriad_plugin.cpp:92]: Code duplication at line 100, isn't it?

File: inference-engine/src/vpu/common/src/ngraph/query_network.cpp [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                return false;
+            } else {
+                return supportedLayers.count((*layer)->name) != 0;
+            }
+    };
+
+    for (InferenceEngine::details::CNNNetworkIterator itLayer{convertedNetwork.get()};
+             itLayer != InferenceEngine::details::CNNNetworkIterator();
+             itLayer++) {
+            const auto fusedNode = (*itLayer)->getNode();
```
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/common/src/ngraph/query_network.cpp:53]: Some problems with indents

**Review by Maxim-Doronin — APPROVED** [review]
Please, fix this nitpick if you are going to rebase

**Review by ZlobinGM — APPROVED** [review]

File: inference-engine/src/vpu/common/src/ngraph/query_network.cpp (hunk 3 above)
Comment (Maxim-Doronin) [reply inference-engine/src/vpu/common/src/ngraph/query_network.cpp:53]: The line `const auto fusedNode = (*itLayer)->getNode();` should have indents as 4 spaces, not 8

File: inference-engine/src/vpu/common/src/ngraph/query_network.cpp [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+    std::unordered_set<std::string> splitNames;
+    std::unordered_set<std::string> concatNames;
+
+    ngraph::NodeVector splits;
+    ngraph::NodeVector concats;
+
+    const auto isLayerSupported = [&supportedLayers, &splitNames, &concatNames, &concats, &splits]
+                                  (InferenceEngine::details::CNNNetworkIterator& layer) -> bool {
+            auto node = (*layer)->getNode();
```
Comment (Maxim-Doronin) [inline inference-engine/src/vpu/common/src/ngraph/query_network.cpp:36]: The same issue with indents

**Review by Maxim-Doronin — APPROVED** [review]

File: inference-engine/src/vpu/common/src/ngraph/query_network.cpp (hunk 4 above)
Comment (adudchen) [reply inference-engine/src/vpu/common/src/ngraph/query_network.cpp:36]: Sure, I'm a little confused :)

**Pull request merged**